### PR TITLE
Update install

### PR DESCRIPTION
**Pull request by juraskov**
I (and @danielhollas) updated install.sh to use mamba instead of conda to speed up installations and pytests. We also updated install_ase.sh to update ase to a specific dev version since ase-master is updated very often.

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -22,18 +22,17 @@ jobs:
...
+    - uses: actions/checkout@v4
         
-    - uses: conda-incubator/setup-miniconda@v2
+    - uses: mamba-org/setup-micromamba@v1
       with:
-        python-version: ${{ matrix.python-version }}
-        channels: conda-forge,defaults
-        auto-update-conda: true
+        # the create command looks like this:
+        # `micromamba create -n test-env python=3.10 numpy`
```
Comment (danielhollas) [inline .github/workflows/pytest.yml:30]:
```suggestion
        # `micromamba create -n test-env python=3.9`
```

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -22,18 +22,17 @@ jobs:
...
+        environment-name: test-env
+        create-args: >-
+          python=3.9
```
Comment (danielhollas) [inline .github/workflows/pytest.yml:33]:
```suggestion
          python=${{ matrix.python-version }}
```

**Review by Hanwen1018 — APPROVED** [review]

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -22,18 +22,17 @@ jobs:
...
         
-    - uses: conda-incubator/setup-miniconda@v2
+    - uses: mamba-org/setup-micromamba@v1
       with:
-        python-version: ${{ matrix.python-version }}
-        channels: conda-forge,defaults
-        auto-update-conda: true
+        # the create command looks like this:
+        # `micromamba create -n test-env python=3.10 numpy`
+        environment-name: test-env
```
Comment (danielhollas) [inline .github/workflows/pytest.yml:31]:
```suggestion
        environment-name: test-env
        cache-environment: true
```

**Review by danielhollas — CHANGES_REQUESTED** [review]
Some tweaks.

**Review by danielhollas — APPROVED** [review]

**Pull request merged**